### PR TITLE
Add unit tests for notification security rules. Closes #4190

### LINKED ISSUE #4190: Unit Test the Notification Security Rules.
Unit Test the Notification Security Rules.

### PR DESCRIPTION
**Pull request by xmano**
Notification security rules UT.

### CODE REVIEW

File: apps/festival/festival-e2e/src/integration/dashboard/movie-tunnel.spec.ts [hunk 1]
```diff
@@ -293,4 +293,9 @@ describe('User can navigate to the movie tunnel pages start and main.', () => {
     })
   });
 
+  //If 
+  it('Publish the movie to the market', () => {
+    
+  });
```
Comment (phpgeekfr) [inline apps/festival/festival-e2e/src/integration/dashboard/movie-tunnel.spec.ts:299]: Is this one passing ? ;)

File: apps/backend-ops/src/firestore-rules.spec.ts [hunk 2]
```diff
@@ -178,21 +178,89 @@ describe('Users Collection Rules Tests', () => {
 
 });
 
-//TODO: 4190
-describe('Notification Rules Tests', () => {
+describe.only('Notification Rules Tests', () => {
```
Comment (phpgeekfr) [inline apps/backend-ops/src/firestore-rules.spec.ts:181]: remove only

File: apps/backend-ops/src/firestore-rules.spec.ts [hunk 3]
```diff
@@ -318,7 +386,7 @@ describe('Movies Rules Tests', () => {
       ["_meta", { createdAt: ''}],
       ["_type", 'drama'],
       ["storeConfig", { appAccess: {catalog: true}}],
-      ["storeConfig", { status: 'rejected'}], //<-- doesn't work??!!
+      ["storeConfig", { status: 'rejected'}],
```
Comment (phpgeekfr) [inline apps/backend-ops/src/firestore-rules.spec.ts:389]: not it this PR scope but :
```
Movies Rules Tests › With User in org › user valid org, with create permission for org should be able to create movie
    FirebaseError: 7 PERMISSION_DENIED: 
    false for 'create' @ L22, Property storeConfig is undefined on object. for 'create' @ L292
```
I removed skip from `test("user valid org, with create permission for org should be able to create movie", async () => {`

File: apps/backend-ops/src/firestore-rules.spec.ts [hunk 4]
```diff
@@ -178,21 +178,89 @@ describe('Users Collection Rules Tests', () => {
```
Comment (phpgeekfr) [inline apps/backend-ops/src/firestore-rules.spec.ts:178]: format file for better readability

**Review by phpgeekfr — APPROVED** [review]
LGTM

File: apps/backend-ops/src/firestore-rules.spec.ts (hunk 2 above)
Comment (xmano) [reply apps/backend-ops/src/firestore-rules.spec.ts:181]: done

File: apps/festival/festival-e2e/src/integration/dashboard/movie-tunnel.spec.ts (hunk 1 above)
Comment (xmano) [reply apps/festival/festival-e2e/src/integration/dashboard/movie-tunnel.spec.ts:299]: from other work.. context switch .. will drop.

File: apps/backend-ops/src/firestore-rules.spec.ts (hunk 3 above)
Comment (xmano) [reply apps/backend-ops/src/firestore-rules.spec.ts:389]: Fixed.

**Comment by xmano** [comment]
Adding nested describe blocks to differentiate the tests and have shorter test descriptions.

```
  Movies Rules Tests
    With User in org
      Read Movie
        √ should be able to read movie (28ms)
        √ should be able to read movie distribution rights (5ms)
      Create Movie
        √ storestatus other than draft should not be able (8ms)
        √ without create permission for org should not be able (6ms)
        √ with create permission for org, invalid id should not be able (4ms)
        √ with create permission for org should be able (5ms)
      Update Movie
        √ updating restricted 'id' field shouldn't be able (3ms)
        √ updating restricted '_meta' field shouldn't be able (13ms)
        √ updating restricted '_meta' field shouldn't be able (8ms)
        √ updating restricted '_type' field shouldn't be able (13ms)
        √ updating restricted 'storeConfig' field shouldn't be able (7ms)
        √ updating restricted 'storeConfig' field shouldn't be able (8ms)
        √ updating restricted 'storeConfig' field shouldn't be able (12ms)
        √ updating restricted 'storeConfig' field shouldn't be able (13ms)
        √ user valid org, updating unrestricted field should be able (9ms)
    User with admin role
      √ User with admin role in org should be able to delete movie (27ms)
    With User not in org
      √ user without valid org shouldn't be able to read movie title (27ms)
      √ user without valid org shouldn't be able to create movie title (10ms)
      √ user without valid org shouldn't be able to delete movie title (4ms)
```

image

File: apps/backend-ops/src/firestore-rules.spec.ts (hunk 4 above)
Comment (xmano) [reply apps/backend-ops/src/firestore-rules.spec.ts]: done

File: apps/backend-ops/src/firestore-rules.spec.ts [hunk 5]
```diff
@@ -388,117 +463,108 @@ describe.skip('Contracts Rules Tests', () => {
...
+      const eventRef = db.doc('events/E001');
       await assertFails(eventRef.delete());
     });
   });
-
 });
 
 //TODO: 4187
 describe.skip('Campaign Security Rules', () => {
+  beforeAll(async () => {});
```
Comment (phpgeekfr) [inline apps/backend-ops/src/firestore-rules.spec.ts:567]: is it needed ?

File: apps/festival/festival-e2e/src/integration/dashboard/movie-tunnel.spec.ts (hunk 1 above)
Comment (phpgeekfr) [reply apps/festival/festival-e2e/src/integration/dashboard/movie-tunnel.spec.ts:299]: still there ?

File: apps/backend-ops/src/firestore-rules.spec.ts [hunk 6]
```diff
@@ -218,14 +292,12 @@ describe.skip('Organization Rules Tests', () => {
...
+    db = await initFirestoreApp(projectId, 'firestore.rules', testFixture, { uid: 'uid-user2' });
   });
 
-  afterAll(() => Promise.all(apps().map(app => app.delete())));
-
-  test("test", async () => {
+  afterAll(() => Promise.all(apps().map((app) => app.delete())));
 
-  });
+  test('test', async () => {});
```
Comment (phpgeekfr) [inline apps/backend-ops/src/firestore-rules.spec.ts:300]: can you remove ?

File: apps/backend-ops/src/firestore-rules.spec.ts [hunk 7]
```diff
@@ -234,17 +306,14 @@ describe.skip('Permission Rules Tests', () => {
...
```
Comment (phpgeekfr) [inline apps/backend-ops/src/firestore-rules.spec.ts:314]: can you remove ?

Comment (xmano) [reply apps/backend-ops/src/firestore-rules.spec.ts:314]: ok

File: apps/backend-ops/src/firestore-rules.spec.ts (hunk 6 above)
Comment (xmano) [reply apps/backend-ops/src/firestore-rules.spec.ts:300]: ok

File: apps/backend-ops/src/firestore-rules.spec.ts (hunk 5 above)
Comment (xmano) [reply apps/backend-ops/src/firestore-rules.spec.ts:567]: these tests are skipped.. was placeholder.

will remove it.

File: apps/festival/festival-e2e/src/integration/dashboard/movie-tunnel.spec.ts (hunk 1 above)
Comment (xmano) [reply apps/festival/festival-e2e/src/integration/dashboard/movie-tunnel.spec.ts:299]: removed. sorry.

File: apps/backend-ops/src/firestore-rules.spec.ts (hunk 6 above)
Comment (xmano) [reply apps/backend-ops/src/firestore-rules.spec.ts]: also remove movie-tunnel stuff.

**Pull request merged**